File: AUGUST/KINHYEONJI/BOJ1110.cpp
#include <iostream>

using namespace std;

int main() {
    int num;
    cin >> num;
    int count = 0;
    int firstNum = num;
    while (true) {
        int newNum = num % 10 * 10 + (num / 10 + num % 10) % 10;
        count++;
        if (newNum == firstNum) {
            cout << count;
            return 0;
        }
        num = newNum;
    }
    return 0;
}

File: AUGUST/shlee/BOJ1110.py
N = int(input())

a = N // 10  # 십의 자리
b = N % 10  # 일의 자리

count = 0       # 사이클 길이 초기값
while True:
    # 십의 자리 -> 원래의 일의 자리 , 일의 자리 -> 두 수의 합의 일의 자리
    a, b = b, (a + b) % 10      
    if (a == N // 10) and (b == N % 10):    # 초기 값으로 돌아오면 반복문 종료
        count += 1
        break
    count += 1      # 반복 사이클 길이 구하기

print(count)